sale.py

```diff
@@ -185,6 +185,12 @@ def on_change_quantity(self):
         if self.unit_price is not None:
             self.update_prices()
 
+    @fields.depends('type', 'taxes')
```

Comment:
- Replace "taxes" by "gross_unit_price" in depends.
- Set "discount" to None in case line != 'type' (same as "gross_unit_price").

Reply:
Agree on setting discount to None, also update_prices() should be called because "gross_unit_price" is not being correctly cleared, but why adding it in depends instead of taxes?
In one hand taxes is added to depends in core sale module, so and I don't think its a good idea to remove it. 
In the other hand, I do not see the point of adding gross_unit_price to it.

Reply:
> Agree on setting discount to None, also update_prices() should be called because "gross_unit_price" is not being correctly cleared, but why adding it in depends instead of taxes?

I think call update_prices()  not set none discount price and finally call "on_change_with_amount"

> In one hand taxes is added to depends in core sale module, so and I don't think its a good idea to remove it.
> In the other hand, I do not see the point of adding gross_unit_price to it.

trytond append the depends in all inherit methods. Example:

- Core module add:

@depend('d1')

- Module A add:

@depend('d2')

- Module B add:

@depend('d3')

In case you install core module + module A, depends are 'd1' and 'd2'
In case you install core module + module B, depends are 'd1' and 'd3'
In case you install core module + module B + module C, depends are 'd1', 'd2' and 'd3'

Try to pass the tests; in case depends are missing, do asset error.

Reply:
> Sorry I don't understand what do you mean, I don't know what should I do.

See [update_prices()](https://github.com/NaN-tic/trytond-sale_discount/blob/ed0573586bb7cbda36eeab3051afdc45b415458c/sale.py#L106) , not set discount to null.

Reply:
Sorry, I still don't understand whay do you mean by "not set discount to null", feel free to update the pull-request by yourself.

Reply:
- changeset 105ceba89a9fcbb70549394591a460b7600f868d 
- Updated test scenario.